firedrake/variational_solver.py

```diff
@@ -95,7 +95,18 @@ def dm(self):
 class NonlinearVariationalSolver(OptionsManager):
     r"""Solves a :class:`NonlinearVariationalProblem`."""
 
-    def __init__(self, problem, **kwargs):
+    DEFAULT_SNES_PARAMETERS = {"snes_type": "newtonls"}
```

Comment:
I think we should also default to `"snes_monitor": None` (in case it's not the default) and `"snes_linesearch_type": "basic"` (for compatibility with FEniCS; the PETSc `"bt"` default breaks lots of codes that otherwise would work).

Reply:
I'm happy to add the linesearch change. The monitoring is a more controversial one: PETSc doesn't turn on monitors by default, although FEniCS does, Firedrake presently doesn't, and this may annoy existing users. Additionally defaulting to `"snes_monitor": None` has the unfortunate side-effect that users will struggle to turn it off: they need to remove it from the dictionary somehow, but if we're shoving it in via defaults...

Reply:
I think they can turn it off with `"snes_monitor_cancel": None`?

Reply:
Oh, that only works if the monitor has been set up in code, not via options. (Maybe that's the way to go, then.)